llama-index-integrations/vector_stores/llama-index-vector-stores-azurecosmosnosql/llama_index/vector_stores/azurecosmosnosql/base.py

```diff
@@ -325,25 +325,31 @@ def _query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResu
 
         # If limit_offset_clause is not specified, add TOP clause
         if pre_filter is None or pre_filter.get("limit_offset_clause") is None:
-            query += "TOP @limit "
+            # query += "TOP @limit "
+            query += f"TOP {params.get('k', 2)} "
 
+        # query += (
+        #     "c.id, c.{}, c.text, c.metadata, "
+        #     "VectorDistance(c.@embeddingKey, @embeddings) AS SimilarityScore FROM c"
+        # )
         query += (
-            "c.id, c.{}, c.text, c.metadata, "
-            "VectorDistance(c.@embeddingKey, @embeddings) AS SimilarityScore FROM c"
+            "c.id, c.text, c.metadata, "
+            f"VectorDistance(c.{self._embedding_key}, @embeddings) AS SimilarityScore FROM c"
         )
 
         # Add where_clause if specified
         if pre_filter is not None and pre_filter.get("where_clause") is not None:
             query += " {}".format(pre_filter["where_clause"])
 
-        query += " ORDER BY VectorDistance(c.@embeddingKey, @embeddings)"
+        # query += " ORDER BY VectorDistance(c.@embeddingKey, @embeddings)"
+        query += f" ORDER BY VectorDistance(c.{self._embedding_key}, @embeddings)"
 
         # Add limit_offset_clause if specified
         if pre_filter is not None and pre_filter.get("limit_offset_clause") is not None:
             query += " {}".format(pre_filter["limit_offset_clause"])
         parameters = [
-            {"name": "@limit", "value": params["k"]},
-            {"name": "@embeddingKey", "value": self._embedding_key},
+            # {"name": "@limit", "value": params["k"]},
+            # {"name": "@embeddingKey", "value": self._embedding_key},
             {"name": "@embeddings", "value": params["vector"]},
```

Comment:
params["vector"] is not the same as self._embedding_key? looking at line 316 above

```
params: Dict[str, Any] = {
            "vector": query.query_embedding,
            "path": self._embedding_key,
            "k": query.similarity_top_k,
        }
```

Reply:
you cannot use the parameter **@embeddingKey** for container field name in the query. in the query we need to use clear field name **c.embeding** instead of **c.@embeddingKey**

@limit will process default 2 if there is no value